### DISCUSSION TITLE
Question about Supertrend and Renko: Is it possible to add a bar without recalculating the entire historical data?

### DISCUSSION THREAD

**Discussion by contrerasdaniel142** [DaveSkender/Stock.Indicators#1090] [Q&A]
Hello everyone,

I'm interested in using the Supertrend and Renko indicators in my financial data analysis, and I have a question about how they work. Is there a way to add an additional bar without having to recalculate the entire historical data each time you want to update these indicators?

Often, updating the entire historical dataset can be a time and resource-intensive process, especially when working with a large amount of data. Therefore, I would like to know if there is any technique or approach that allows adding a new data bar without having to recalculate the entire history.

I appreciate any information or advice you can provide. This would be a great help in improving my efficiency in technical analysis. Thank you!

**Comment by DaveSkender** [comment]
> Is there a way to add an additional bar without having to recalculate the entire historical data each time you want to update these indicators?

Not yet, but we're working on that use case next.

Follow or continue this conversation in our pre-existing [discussion on streaming indicators](https://github.com/DaveSkender/Stock.Indicators/discussions/1018) for progress and information about pre-release package versions to support live and incremental quote sources.  This is the next major feature area in our [roadmap](https://github.com/DaveSkender/Stock.Indicators/discussions/218)
